src/main/java/com/newrelic/agent/instrumentation/compatibility/CompatibilitySiteTask.java

```diff
@@ -57,6 +58,7 @@ public void setUrl(String url) {
         this.url = url;
     }
 
+    @Input
```

Comment:
Can someone confirm that these are the proper annotation types? https://docs.gradle.org/6.3/userguide/more_about_tasks.html#sec:task_input_output_annotations

Reply:
Do we have a gradle expert that might actually be able to answer this!?  

The docs say to annotate the outputs as well, but I think this is so confusing that I can't say anything about this with any degree of confidence.  That link also talks about _incremental_ build support...is that what you were going for?  Did you test it or is it even testable?

Reply:
The annotation seems right but I don't understand why we have them?   

I read the doc (as much as that doc can be read...) -  My understanading is Input and Output annotations are not necessary. They are nice to have for the build process and offer enhancements. You'll get warnings when the task is run.

If I'm right with that, what are we seeking to gain with these annotations?

Reply:
Here is one test: 

Using this branch, I published the release version to Maven Local. The Java agent consumed it and the plugin worked. I was able to generate the output files.

Reply:
So I added these annotations because the `./gradlew validatePlugins` fails without them which causes the build to fail. It seems that this is part of the recommended upgrade process from gradle 5 to 6. https://docs.gradle.org/current/userguide/upgrading_version_5.html

Technically it just produces warnings which are enough to fail the task.

```
* What went wrong:
Execution failed for task ':validatePlugins'.

> Plugin validation failed. See https://docs.gradle.org/6.3/userguide/more_about_tasks.html#sec:task_input_output_annotations for more information on how to annotate task properties.

   > Warning: Type 'CompatibilitySiteTask': property 'documentation' is not annotated with an input or output annotation.
   > Warning: Type 'CompatibilitySiteTask': property 'title' is not annotated with an input or output annotation.
   > Warning: Type 'CompatibilitySiteTask': property 'type' is not annotated with an input or output annotation.
   > Warning: Type 'CompatibilitySiteTask': property 'url' is not annotated with an input or output annotation.
```

Reply:
@breedx-nr   It looks like incremental build is a feature that became default in Gradle 5 and has been improved upon in every release after.  It doesn't appear to be something one "opts in" for.

Reply:
It looks like we have annotated the output here...

https://github.com/newrelic/newrelic-gradle-compatibility-doc-plugin/blob/97b1430c2615585f916f18e2b2202f1fb021e847/src/main/groovy/com/newrelic/agent/instrumentation/compatibility/CompatibilitySitePlugin.groovy#L22

In manual testing, the lack of these annotations don't break `validatePlugin` or the build. 
Also, it probably should be @OutputFile here

```
@OutputFile
File jsonOutput
```
I don't think this class is the right place for the Output annotation. From the doc example, it should be in `CompatibilitySiteTask`. 

However, we don't need output annotation because we don't have getters for our outputs.